datafusion/core/src/datasource/file_format/csv.rs

```diff
@@ -385,6 +387,46 @@ mod tests {
         Ok(())
     }
 
+    #[tokio::test]
+    async fn infer_shcmea_with_limit() {
+        let session = SessionContext::new();
+        let ctx = session.state();
+        let store = Arc::new(LocalFileSystem::new()) as _;
+        let filename = "tests/csv/schema_infer_limit.csv";
+        let format = CsvFormat::default().with_schema_infer_max_rec(Some(3));
+
+        let file_schema = format
+            .infer_schema(&ctx, &store, &[local_unpartitioned_file(filename)])
+            .await
+            .expect("Schema inference");
```

Comment:
Can you explain why you chose to add this test? I think this feature is already covered by existing tests.

I see the test `infer_schema` above this that has a limit of `Some(1)`.

Which should call infer schema as part of its execution.